WordPress/gutenberg
# Comment Template block: Handle nested comments

## Description
This is a follow-up PR to https://github.com/WordPress/gutenberg/pull/35231 which adds handling of nested comments.

## How has this been tested?

1. Install and activate a FSE theme (like the "TT1 Blocks" ).
2. Create a new post.
3. Add some comments to the new post, including some **nested** comments.
4. Edit the post and insert the `Comments Query Loop` block .
5. Check that the replies are indeed nested under the their respective "parent" comments.

## Screenshots

### admin
<img width="675" alt="Screen Shot 2021-11-25 at 20 06 30" src="https://user-images.githubusercontent.com/5417266/143512292-23e1ef22-4e03-4eb7-a894-76d0526f9c68.png">

### frontend
<img width="537" alt="Screen Shot 2021-11-25 at 20 07 06" src="https://user-images.githubusercontent.com/5417266/143512326-7073f2b6-b7bb-4c31-ae4d-7054e48bb004.png">

## Types of changes

This is still a pretty barebones implementation of comment nesting. 

Styling is very basic and will be tackled in another PR. For now, I've just included enough so that you can visually discern the nested comments by their indentation.

Also:
- The list of comments is now a `ol` instead of `ul` element because comments are an **ordered list**
- There are still some glitches in the UI as you attempt to navigate / switch focus between the contents of the Comment Template - sometimes the blocks do not get "active" immediately. Will also handle it in another PR.
- I've removed the code that was supposed to inherit settings from the Discussion settings that was present in the early versions of this PR for reasons described below in https://github.com/WordPress/gutenberg/pull/36065#issuecomment-955097061
